### PR TITLE
Replace ppu_module_manager Function Static with Class Static variable (static module map)

### PR DESCRIPTION
**Pull request by bevanweiss**
Makes for a slightly 'cleaner' interface in my opinion, may also assist with adding thread read/write concurrency support in future if ever required (have left that out of this commit to match existing function).

Very slight performance improvements were seen in representative testing.
https://quick-bench.com/q/GMbgeNc-mZc21aqOKCofnbzPZvg

I didn't investigate whether static initialisation of the static_modules might actually be possible here, perhaps there's a way to do a constexpr / consteval of this.

### CODE REVIEW

**Comment by bevanweiss** [comment]
I'll drop the auto on the get() and have the implicit cast instead.

File: rpcs3/Emu/Cell/PPUModule.h [hunk 1]
```diff
@@ -273,6 +276,9 @@ class ppu_module_manager final
 	static const ppu_static_module sys_libc;
 	static const ppu_static_module sys_lv2dbg;
 	static const ppu_static_module static_hle;
+
+private:
+	inline static std::unordered_map<std::string, ppu_static_module*> staticModuleMap;
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.h:281]: we prefer snake_case over CamelCase especially in places where the rest of the code is already snake_case.

File: rpcs3/Emu/Cell/PPUModule.h [hunk 2]
```diff
@@ -101,7 +102,8 @@ class ppu_module_manager final
 {
 	friend class ppu_static_module;
 
-	static std::unordered_map<std::string, ppu_static_module*>& access();
+	// this is now replaced by the staticModuleMap static member variable
+	//static std::unordered_map<std::string, ppu_static_module*>& access();
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.h:106]: remove lines.

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 3]
```diff
@@ -99,7 +100,7 @@ ppu_static_variable& ppu_module_manager::access_static_variable(const char* _mod
 
 const ppu_static_module* ppu_module_manager::get_module(const std::string& name)
 {
-	const auto& map = access();
+	const auto& map  = ppu_module_manager::staticModuleMap;
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.cpp:103]: wild space before =

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 4]
```diff
@@ -87,7 +88,7 @@ ppu_static_function& ppu_module_manager::access_static_function(const char* _mod
 
 ppu_static_variable& ppu_module_manager::access_static_variable(const char* _module, u32 vnid)
 {
-	auto& res = access().at(_module)->variables[vnid];
+	auto& res = ppu_module_manager::staticModuleMap.at(_module)->variables[vnid];
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.cpp:91]:
```suggestion
	auto& res = staticModuleMap.at(_module)->variables[vnid];
```

Comment (bevanweiss) [reply rpcs3/Emu/Cell/PPUModule.cpp:91]: You don't feel that leaving in the full scoping makes it 'more' apparent that it's a static member variable and not just 'any old' member variable?

I'll fix up the others, and prepare to push again (and will just wait for your feedback on this one)

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 5]
```diff
@@ -61,21 +61,22 @@ ppu_static_module::ppu_static_module(const char* name)
 	ppu_module_manager::register_module(this);
 }
 
-std::unordered_map<std::string, ppu_static_module*>& ppu_module_manager::access()
-{
-	static std::unordered_map<std::string, ppu_static_module*> map;
-
-	return map;
-}
+// We move the static unordered_map up into the ppu_module_manager definition
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/PPUModule.cpp:64]: What's the use of this commented code?

Comment (bevanweiss) [reply rpcs3/Emu/Cell/PPUModule.cpp:64]: I thought it was staying inline with the 'Comment every hack you do, every snippet you comment out and every improvable code.' (i.e. the every snippet you comment out)
If you'd prefer it just gone, I can do that.

**Comment by kd-11** [comment]
Is this code change justifiable though?
By that I mean, it obviously doesn't help the users, which then leaves the issue of whether it helps developers cut down on time or not. Otherwise a few months from now someone else comes along and adjusts the same code to a style they prefer which is obviously not a good idea.
From a programming POV, this just replaces access() with a raw variable name (that is also longer to type but that's not really an big deal) so I'm left scratching my head why this is necessary. Remember that every code change made undoes a contribution by someone else, modifies project history, etc so there is a fixed negative cost associated with it. It must at least offer some objective advantage resource-wise to counteract this. If for example it makes the user experience better it is a no-brainer to include as the negative costs cited are justified.
Now, the code change isn't really in a part of the project I spend much time in, that would be for Neko or Elad to decide, but this is just something to keep in mind. If we don't have some kind of restraint the whole codebase would be rewritten every couple of years with "new" coding styles in a never-ending deprecation cycle.

File: rpcs3/Emu/Cell/PPUModule.h [hunk 6]
```diff
@@ -273,6 +272,9 @@ class ppu_module_manager final
 	static const ppu_static_module sys_libc;
 	static const ppu_static_module sys_lv2dbg;
 	static const ppu_static_module static_hle;
+
+private:
+	inline static std::unordered_map<std::string, ppu_static_module*> static_module_map;
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.h:277]:
```suggestion
	inline static std::unordered_map<std::string, ppu_static_module*> s_module_map;
```
we prefer the prefix `s_` for static variables, its mentioned in our code style guidelines.

Comment (bevanweiss) [reply rpcs3/Emu/Cell/PPUModule.h]: Ofcourse github doesn't do inbuilt refactoring...
I'll fix the name shortly

**Pull request merged**